File: app/src/main/java/com/delarosa/cognitoAuth/model/SharePreferences.java
package com.delarosa.cognitoAuth.model;

import android.content.Context;
import android.content.SharedPreferences;

public class SharePreferences {

    private  Context context;

    public SharePreferences(Context context){
        this.context = context;
    }

    public final static String PREFS_NAME = "preferenciasApp";

    public  void setInteger(String key, int value) {
        SharedPreferences sharedPref = context.getSharedPreferences(PREFS_NAME,0);
        SharedPreferences.Editor editor = sharedPref.edit();
        editor.putInt(key, value);
        editor.apply();
    }

    public  int getInteger(String key) {
        SharedPreferences prefs = context.getSharedPreferences(PREFS_NAME, 0);
        return prefs.getInt(key, 0);
    }

    public  void setString(String key, String value) {
        SharedPreferences sharedPref = context.getSharedPreferences(PREFS_NAME,0);
        SharedPreferences.Editor editor = sharedPref.edit();
        editor.putString(key, value);
        editor.apply();
    }

    public  String getString(String key) {
        SharedPreferences prefs = context.getSharedPreferences(PREFS_NAME, 0);
        return prefs.getString(key,"DNF");
    }

    public  void setBoolean(String key, boolean value) {
        SharedPreferences sharedPref = context.getSharedPreferences(PREFS_NAME,0);
        SharedPreferences.Editor editor = sharedPref.edit();
        editor.putBoolean(key, value);
        editor.apply();
    }

    public  boolean getBoolean(String key) {
        SharedPreferences prefs = context.getSharedPreferences(PREFS_NAME, 0);
        return prefs.getBoolean(key,false);
    }
}

File: app/src/main/java/com/delarosa/cognitoAuth/model/SharePreferences.kt
package com.delarosa.cognitoAuth.model

import android.content.Context
import android.content.SharedPreferences

class SharePreferences(private val context: Context) {

    fun setInteger(key: String, value: Int) {
        val sharedPref = context.getSharedPreferences(PREFS_NAME, 0)
        val editor = sharedPref.edit()
        editor.putInt(key, value)
        editor.apply()
    }

    fun getInteger(key: String): Int {
        val prefs = context.getSharedPreferences(PREFS_NAME, 0)
        return prefs.getInt(key, 0)
    }

    fun setString(key: String, value: String) {
        val sharedPref = context.getSharedPreferences(PREFS_NAME, 0)
        val editor = sharedPref.edit()
        editor.putString(key, value)
        editor.apply()
    }

    fun getString(key: String): String {
        val prefs = context.getSharedPreferences(PREFS_NAME, 0)
        return prefs.getString(key, "DNF")
    }

    fun setBoolean(key: String, value: Boolean) {
        val sharedPref = context.getSharedPreferences(PREFS_NAME, 0)
        val editor = sharedPref.edit()
        editor.putBoolean(key, value)
        editor.apply()
    }

    fun getBoolean(key: String): Boolean {
        val prefs = context.getSharedPreferences(PREFS_NAME, 0)
        return prefs.getBoolean(key, false)
    }

    companion object {

        val PREFS_NAME = "preferenciasApp"
    }
}